libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -0,0 +1,817 @@
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
+    if( TEST_PROTECT() )
+    {
+        lWrite = iot_uart_write_sync( xUartHandle, cpBuffer, testIotUART_BUFFER_LENGTH );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lWrite );
+
+        lRead = iot_uart_read_sync( xUartHandle, cpBufferRead, testIotUART_BUFFER_LENGTH );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lRead );
+
+        TEST_ASSERT_EQUAL( 0, strncmp( ( char * ) cpBuffer, ( char * ) cpBufferRead, testIotUART_BUFFER_LENGTH ) );
+    }
+
+    lClose = iot_uart_close( xUartHandle );
+    TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lClose );
+}
+/*-----------------------------------------------------------*/
+/*-----------------------------------------------------------*/
+/**
+ * @brief Test Function to check UART Ioctl API specifications.
+ */
+TEST( TEST_IOT_UART, AFQP_IotUARTIoctlGetSet )
+{
+    IotUARTHandle_t xUartHandle;
+    int32_t lIoctl, lClose, lTransferAmount;
+    IotUARTConfig_t pvgetConfigBuffer, pvConfigBuffer, pvOriginalConfig;
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
+    if( TEST_PROTECT() )
+    {
+
+        /* Save original UART config */
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvOriginalConfig );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_NOT_EQUAL( testIotUART_TEST_BAUD_RATE, pvOriginalConfig.ulBaudrate );
+        TEST_ASSERT_NOT_EQUAL( testIotUART_TEST_BAUD_RATE, pvgetConfigBuffer.ulBaudrate );
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvConfigBuffer );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+
+        /* Test changing baudrate config */
+        pvConfigBuffer.ulBaudrate = testIotUART_TEST_BAUD_RATE;
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartSetConfig, &pvConfigBuffer );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvgetConfigBuffer );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_EQUAL( testIotUART_TEST_BAUD_RATE, pvgetConfigBuffer.ulBaudrate );
+
+        /* Test changing control flow config */
+        pvConfigBuffer.ucFlowControl = ultestIotUartFlowControl;
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartSetConfig, &pvConfigBuffer );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_NOT_EQUAL( ultestIotUartFlowControl, pvOriginalConfig.ucFlowControl );
+        TEST_ASSERT_NOT_EQUAL( ultestIotUartFlowControl, pvgetConfigBuffer.ucFlowControl );
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvgetConfigBuffer );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_EQUAL( ultestIotUartFlowControl, pvgetConfigBuffer.ucFlowControl );
+
+        /* Test changing stop bit config */
+        pvConfigBuffer.xStopbits = ultestIotUartStopBits;
```

Comment:
In the previous test_iot_uart.c, there is a xSampleConfig2 and is used here. It has different configuration. Why is it taken away?

Reply:
This PR is about a month old, and test_iot_uart.c  has had some changes and been rebased. I think we should scrap this PR and rebase the python scripts onto the latest version.